features/admin.applications.v1/components/forms/inbound-oidc-form.tsx

```diff
@@ -435,6 +457,74 @@ export const InboundOIDCForm: FunctionComponent<InboundOIDCFormPropsInterface> =
         );
     }, [ application ]);
 
+    const fetchLocalClaims: () => void = useCallback(() => {
+        getAllLocalClaims(null)
+            .then((response: Claim[]) => {
+                setClaims(response);
+            })
+            .catch(() => {
+                dispatch(addAlert({
+                    description: t("claims:local.notifications.fetchLocalClaims.genericError.description"),
+                    level: AlertLevels.ERROR,
+                    message: t("claims:local.notifications.fetchLocalClaims.genericError.message")
+                }));
+            });
+    }, []);
+
+    const fetchExternalClaims: () => void = useCallback(() => {
+        getAllExternalClaims(OIDCScopesManagementConstants.OIDC_ATTRIBUTE_ID, null)
+            .then((response: ExternalClaim[]) => {
+                setExternalClaims(response);
+            })
+            .catch(() => {
+                dispatch(addAlert({
+                    description: t("claims:external.notifications.fetchExternalClaims.genericError.description"),
+                    level: AlertLevels.ERROR,
+                    message: t("claims:external.notifications.fetchExternalClaims.genericError.message")
+                }));
+            });
+    }, []);
```

Comment:
Let's move this data fetching and related logic as well to the new component defined.

Reply:
new component is for single attribute. moving this block to component seems wrong? wdyt

Reply:
Also, why these functions are `useCallback` ?

Reply:
will remove `useCallback`